### PR TITLE
Friendship v2

### PR DESCRIPTION
**Pull request by themonster2015**
Hi,

Please check and approve my submission. Thank you!

### CODE REVIEW

File: app/views/users/_user.html.erb [hunk 1]
```diff
@@ -7,12 +7,12 @@
...
-  <% elsif current_user.friend_requests.include?(user) %>
+  <% elsif current_user.requested_friends.include?(user) %>
     <div class="d-flex justify-content-end">
       <%= button_to 'Accept', user_accept_path(user), class: 'btn btn-primary mr-2'%>
       <%= button_to 'Reject', user_unfriend_path(user, format: :reject_request ), method: :delete, class: 'btn btn-danger'%>
     </div>
-  <% elsif !current_user.friend?(user) %>
+  <% elsif !current_user.friends_with?(user) %>
     <%= button_to 'Send friend request', user_friend_path(user), class: 'btn btn-primary'%>
   <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/_user.html.erb:17]: Put this logic in a view helper in app/helpers and render it here.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] Commit messages are descriptive.
- [ ] Project UI has very good features.
## To Improve 🙏
- [x] There should be a main/master branch that is used for production releases.
- [x] PR description should include the information about features of the project.
- [x] README should include `rails db:create` step before `rails db:migrate`.
- [ ] Please add Github actions so that linter checks can show on PR.
- [x] `AddDefaultValueToFriendshipConfirmedMigrations` migration is failed to run. Kindly fix it.
- [x] There is no option on the navigation to send friendship requests to other users.
- [x] User is unable to see all users with the Friend Request button next to users who are not their friends already.
- [x] This error occurs when a user has friendship requests and he clicks on the notifications button.
notification
- [x] User is unable to accept incoming friendship requests. Please see this error.
friendship_error
- [x] User is unable to reject incoming friendship requests. Please see this error.
friendship_reject_error
- [x] User should be able to see all posts by him and by his friends.
- [ ] Please see inline comments for reference.
# Optional
- [ ] Views should be separate from logic. You can put logic in view helpers under app/helpers and render them in views.
### Happy Coding ☺️
## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Comment by themonster2015** [comment]
first of all, thanks for the comments and feedback
can you explain to me this step: Please add Github actions so that linter checks can show on PR?

also the accept friend request, reject friend request, view posts from friends and self on the dashboard functions work well on my computer

i don't know why they display errors on your side

Screen Shot 2021-06-15 at 22 33 01
Screen Shot 2021-06-15 at 22 26 51
Screen Shot 2021-06-15 at 22 26 39
Screen Shot 2021-06-15 at 22 26 20

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
**_Hello Team_** 👋🏽👋🏽👋🏽
**_Your project is really nice. I like it!!!_** 👏🏽

**Kindly work on this requested changes:**
- Remove this message ```--- !ruby/object:ActionController::Parameters
parameters: !ruby/hash:ActiveSupport::HashWithIndifferentAccess
  controller: posts
  action: index
permitted: false ```
- When I create a post within a new account and then logout and access to a new one i can see the posts from other users.
- If i accept a new friend request the app crashes: 
Screen Shot 2021-06-16 at 06 18 19
- If I reject a friend request the app crashes: 
Screen Shot 2021-06-16 at 06 19 07

**Kindly work on this requested changes and after create a new code review.**
**_Happy Coding!_** 👨🏽‍💻
@carlos-ssh

**Comment by themonster2015** [comment]
Hi,

Thanks very much for your feedback. I've fixed it.

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coder, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The app runs as expected and doesn't crash any more


### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

<hr>

### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Pull request merged**